src/rrdp.rs

```diff
@@ -361,95 +463,234 @@ pub trait ProcessDelta {
+/// all references to RRDP files are given with a SHA-256 hash over the
+/// expected content of that file, allowing a client to verify they got the
+/// right file.
 #[derive(Clone, Debug)]
 pub struct UriAndHash {
+    /// The URI of the RRDP file.
     uri: uri::Https,
-    hash: DigestHex,
+
+    /// The expected SHA-256 hash over the file’s content.
+    hash: Hash,
 }
 
 impl UriAndHash {
-    pub fn new(uri: uri::Https, hash: DigestHex) -> Self {
+    /// Creates a new URI-and-hash pair.
+    pub fn new(uri: uri::Https, hash: Hash) -> Self {
         UriAndHash { uri, hash }
     }
 
+    /// Returns a reference to the URI.
     pub fn uri(&self) -> &uri::Https {
         &self.uri
     }
 
-    pub fn hash(&self) -> &DigestHex {
-        &self.hash
+    /// Returns the expected SHA-256 hash.
+    pub fn hash(&self) -> Hash {
+        self.hash
+    }
+
+    /// Converts the pair into just the URI.
+    pub fn into_uri(self) -> uri::Https {
+        self.uri
+    }
+
+    /// Converts `self` into a pair of URI and hash.
+    pub fn into_pair(self) -> (uri::Https, Hash) {
+        (self.uri, self.hash)
     }
 }
 
 
-//------------ DigestHex -----------------------------------------------------
+//------------ Hash ----------------------------------------------------------
 
-/// A helper type to encode a digest as a sequence of hex-digits.
-#[derive(Clone, Debug)]
-pub struct DigestHex(Vec<u8>);
+/// A hash over RRDP data.
+///
+/// This hash is used both for verifying the correctness of RRDP files as well
+/// as update or deletion of the right objects.
+///
+/// RRDP exclusively uses SHA-256 and provides no means of chosing a different
+/// algorithm. Consequently, this type is essentially a wrapper around a 32
+/// byte array holding SHA-256 output.
+#[derive(Clone, Copy, Eq, hash::Hash, PartialEq)]
+#[repr(transparent)]
+pub struct Hash([u8; 32]);
+
+impl Hash {
+    /// Returns a reference to the octets as a slice.
+    pub fn as_slice(&self) -> &[u8] {
+        self.0.as_ref()
+    }
+}
+
+
+//--- From, TryFrom, and FromStr
 
-impl From<Vec<u8>> for DigestHex {
-    fn from(value: Vec<u8>) -> DigestHex {
-        DigestHex(value)
+impl From<[u8;32]> for Hash {
+    fn from(value: [u8;32]) -> Hash {
+        Hash(value)
     }
 }
 
-impl From<digest::Digest> for DigestHex {
-    fn from(value: digest::Digest) -> DigestHex {
-        DigestHex(Vec::from(value.as_ref()))
+impl From<Hash> for [u8; 32] {
+    fn from(src: Hash) -> Self {
+        src.0
     }
 }
 
-impl str::FromStr for DigestHex {
-    type Err = &'static str;
+impl<'a> TryFrom<&'a [u8]> for Hash {
+    type Error = std::array::TryFromSliceError;
+
+    fn try_from(src: &'a [u8]) -> Result<Self, Self::Error> {
+        TryFrom::try_from(src).map(Hash)
+    }
+}
+
+impl TryFrom<digest::Digest> for Hash {
+    type Error = AlgorithmError;
+
+    fn try_from(digest: digest::Digest) -> Result<Self, Self::Error> {
+        // XXX This doesn’t properly check the algorithm.
```

Comment:
This will be addressed in a follow-up PR which will change `crypto::digest::Digest` from a type alias into ring into its own type including access to the algorithm type.